Sources/ParseSwift/Authentication/3rd Party/ParseFacebook.swift

```diff
@@ -0,0 +1,395 @@
+//
+//  ParseFacebook.swift
+//  ParseSwift
+//
+//  Created by Abdulaziz Alhomaidhi on 3/18/21.
+//  Copyright © 2021 Parse Community. All rights reserved.
+
+import Foundation
+#if canImport(Combine)
+import Combine
+#endif
+
+// swiftlint:disable line_length
+
+/**
+ Provides utility functions for working with Facebook User Authentication and `ParseUser`'s.
+ Be sure your Parse Server is configured for [sign in with Facebook](https://docs.parseplatform.org/parse-server/guide/#configuring-parse-server-for-sign-in-with-facebook).
+ For information on acquiring Facebook sign-in credentials to use with `ParseFacebook`, refer to [Facebook's Documentation](https://developers.facebook.com/docs/facebook-login/limited-login.
+ */
+public struct ParseFacebook<AuthenticatedUser: ParseUser>: ParseAuthentication {
+
+    /// Authentication keys required for Facebook authentication.
+    enum AuthenticationKeys: String, Codable {
+        case id // swiftlint:disable:this identifier_name
+        case authenticationToken
+        case accessToken
+        case expirationDate
+        
+        enum CodingKeys: String, CodingKey {
+          case id // swiftlint:disable:this identifier_name
+          case token
+          case accessToken = "access_token"
+          case expirationDate = "expiration_date"
+        }
+        /// Properly makes an authData dictionary with the required keys.
+        /// - parameter userId: Required id for the user.
+        /// - parameter authenticationToken: Required identity token for the user for Facebook limited login.
+        /// - parameter accessToken: Required identity token for the user for Facebook graph API
+        /// - parameter expirationDate: required expiration data for user authentication for Facebook login
+        /// - returns: Required authData dictionary.
+        func makeDictionary(userId: String?, accessToken: String?, authenticationToken: String?, expirationDate: Date?) throws -> [String: String]? {
+            
+            let validAuthenticationToken = authenticationToken != nil || accessToken != nil
```

Comment:
`validAuthenticationToken` doesn’t seem needed. Can you look at the version of `makeDictionary` I provided to you in the comments. This method doesn’t need to through and can be smaller